packages/headless/src/api/platform-client.ts

```diff
@@ -16,7 +17,8 @@ export interface PlatformClientCallOptions {
   requestParams: unknown;
   accessToken: string;
   renewAccessToken: () => Promise<string>;
-  preprocessRequest: PreprocessRequestMiddleware;
+  preprocessRequest: PreprocessRequest;
+  deprecatedPreprocessRequest: PreprocessRequestMiddleware;
```

Comment:
One concern with the rename to `deprecatedPreprocessRequest`, won't this break existing code using `preprocessRequest`?

Couldn't we instead keep the old name, deprecate, and place the new functionality under a different name?
e.g. `modifyRequest`

Reply:
It should not break because the exposed method is still using the same name, internally it's changed to deprecatedPreprocessRequest